doc/dev/adr/0025-rpc-streaming.md

```diff
@@ -0,0 +1,399 @@
+  - The same MQTT topic as the invoked method
+  - The same correlation data as the invoked method 
+  - Streaming metadata with the ["cancel" flag set](#streaming-user-property)
+  - No payload
+- The command invoker should still listen on the response topic for a response from the executor which may still contain a successful response (if cancellation was received after the command completed successfully) or a response signalling that cancellation succeeded ("Canceled" error code)
+
+As detailed below, the executor may also cancel the stream at any time. In response to receiving a cancellation request from the executor, the invoker should send an MQTT message with:
+ - The same topic as the command itself
+ - The same correlation data as the command itself
+ - The "Canceled" error code
+
+After receiving an acknowledgement from the executor side that the stream has been canceled, any further received messages should be acknowledged but not given to the user.
+
+#### Executor side
+
+Upon receiving an MQTT message with the stream "cancel" flag set to "true" that correlates to an actively executing streaming command, the command executor should:
+ - Notify the application layer that that RPC has been canceled if it is still running
+ - Send an MQTT message to the appropriate response topic with error code "canceled" to notify the invoker that the RPC has stopped and no further responses will be sent.
+
+If the executor receives a cancellation request for a streaming command that has already completed, then the cancellation request should be ignored.
+
+The executor may cancel receiving a stream of requests or cancel sending a stream of responses as well. It does so by sending an MQTT message to the invoker with:
+  - The same MQTT topic as command response
+  - The same correlation data as the invoked method 
+  - Streaming metadata with the ["cancel" flag set](#streaming-user-property)
+  - No payload
+
+The command invoker should then send a message on the same command topic with the same correlation data with the "stream canceled successfully" flag set.
+
+Any received MQTT messages pertaining to a command that was already canceled should still be acknowledged. They should not be given to the user, though.
+
+### Disconnection scenario considerations
+
+- Invoker side disconnects unexpectedly while sending requests
+  - Upon reconnection, the request messages queued in the session client should send as expected
+  - If no reconnection, the streaming RPC will timeout
+- Invoker side disconnects unexpectedly while receiving responses
+  - The broker should hold all published responses for as long as the invoker's session lives and send them upon reconnection
+  - If the invoker's session is lost, then the RPC will timeout
+- Executor side isn't connected when invoker sends first request
+  - Invoker will receive a "no matching subscribers" puback
```

Comment:
this is broker dependent behavior, so we can't count on receiving this puback (some brokers will just return a successful puback)